lte/gateway/c/core/oai/tasks/amf/amf_client_servicer.cpp

```diff
@@ -0,0 +1,57 @@
+/**
+ * Copyright 2020 The Magma Authors.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "include/amf_client_servicer.h"
+#include <memory>
+
+namespace magma5g {
+
+AmfClientServicer amf_client_servicer_g;
```

Comment:
The overall pattern looks more complex than it needs to be. You are wrapping a singleton class with another class and then instantiating a global instance. Instead of this, it makes more sense to convert singleton class into a non singleton one and define a global. Then you can write you mock class directly for that.